SpatialGDK/Source/SpatialGDK/Private/Interop/ActorSystem.cpp

```diff
@@ -798,6 +812,11 @@ void ActorSystem::DestroySubObject(const FUnrealObjectRef& ObjectRef, UObject& O
 	}
 }
 
+void ActorSystem::MarkClientInterestDirty(Worker_EntityId EntityId, bool bOverwrite)
+{
+	ClientInterestDirty.FindOrAdd(EntityId) |= bOverwrite;
+}
```

Comment:
```suggestion
enum class EInterestToWrite
{
	Difference,
	Full
}
void ActorSystem::MarkClientInterestDirty(Worker_EntityId EntityId, EInterestToWrite Interest)
{
	EInterestToWrite& InterestToWrite = ClientInterestDirty.FindOrAdd(EntityId);
	InterestToWrite = InterestToWrite == EInterestToWrite::Full ? EInterestToWrite::Full : Interest;
}
```
Bit more verbose, but `bOverwrite` seems quite ambiguous in this context would prefer an enum.

Reply:
I agree this is more verbose, what about a compromise with a more aptly named bool argument, i.e. `bOnlySendDiff`?

Reply:
Yeah that would be an improvement. Still personally of the opinion that this is a bad use case for a `bool`, especially with this amount of indirection but I can let it go.

Reply:
```suggestion
enum class EInterestToWrite
{
	Difference,
	Full
}

inline EInterestToWrite& operator|= (EInterestToWrite& a, EInterestToWrite b)
{
	return (EInterestToWrite&)((int&)(a) |= (int)(b));
}

void ActorSystem::MarkClientInterestDirty(Worker_EntityId EntityId, EInterestToWrite Interest)
{
	ClientInterestDirty.FindOrAdd(EntityId) |= Interest;
}
```
Alternatively you could do something like this, but maybe too tricksy to add custom operators to an enum.

Reply:
Or you could use unreals `ENUM_CLASS_FLAGS` macro which will define all these operations for you.

Reply:
Is there much indirection going on? It's just setting a bool in a map, and reading it later. I agree the member variable could be named better to indicate the purpose of the bool, likewise with the function parameter, but enum approach seems like overkill. The bitwise enum solution is neat, but also seems somewhat ambiguous, as both flags could be set simultaneously.